apps/electron/src/helper/db/base-db-adapter.ts

```diff
@@ -13,6 +18,10 @@ export abstract class BaseSQLiteAdapter {
 
   async connectIfNeeded() {
     if (!this.db) {
+      const validation = await SqliteConnection.validate(this.path);
+      if (validation === ValidationResult.MissingVersionColumn) {
+        await migrateToLatestDatabase(this.path);
+      }
```

Comment:
there is a _rare_ case that db connection quickly connect then disconnect during this time.
Also I would like to try this change with the corrupted data provided by the users first. brb